bugsnag-plugin-android-performance-appcompat/src/test/kotlin/com/bugsnag/android/performance/FragmentActivityLifecycleCallbacksTest.kt

```diff
@@ -0,0 +1,90 @@
+package com.bugsnag.android.performance
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import com.bugsnag.android.performance.internal.BugsnagPerformanceInternals
+import com.bugsnag.android.performance.internal.SpanFactory
+import com.bugsnag.android.performance.internal.SpanTracker
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class FragmentActivityLifecycleCallbacksTest {
+    private lateinit var spanCollector: MutableList<Span>
+
+    private lateinit var spanTracker: SpanTracker
+
+    private lateinit var lifecycleCallbacks: FragmentActivityLifecycleCallbacks
+
+    private lateinit var fm: FragmentManager
+    private lateinit var fragment1: Fragment
+    private lateinit var fragment2: Fragment
+
+    @Before
+    fun setup() {
+        spanCollector = ArrayList()
+        spanTracker = SpanTracker()
+        lifecycleCallbacks = FragmentActivityLifecycleCallbacks(
+            spanTracker,
+            SpanFactory(spanCollector::add),
+        )
+
+        fm = object : FragmentManager() {}
+        fragment1 = mock {
+            whenever(it.isAdded) doReturn true
+        }
+        fragment2 = mock {
+            whenever(it.isAdded) doReturn true
+        }
+    }
+
+    @After
+    fun shutdown() {
+        BugsnagPerformanceInternals.currentSpanContextStack.clear()
+    }
+
+    /**
+     * Test that when two fragments are being started interleaved within the same transaction,
+     * the SpanContext is what we expect it to be
+     */
+    @Test
+    fun testInterleavedLifecycles() {
```

Comment:
I wonder if its worth making this test a bit stricter? e.g. asserting that the collected spans have the correct parentage and/or `makeContext` value?

Reply:
This requires that we open `SpanImpl.parentSpanId` as `public`, but I think that's a valuable change for our modules as `spanId` and `traceId` are already public